src/Microsoft.OData.Core/Metadata/MetadataUtils.cs

```diff
@@ -171,27 +171,15 @@ internal static IList<IEdmOperation> CalculateBindableOperationsForType(IEdmType
                 throw new ODataException(Strings.MetadataUtils_CalculateBindableOperationsForType(bindingType.FullTypeName()), exc);
             }
 
-            List<IEdmOperation> operationsFound = new List<IEdmOperation>();
-            foreach (IEdmOperation operation in operations)
-            {
-                if (!operation.IsBound)
-                {
-                    throw new ODataException(Strings.EdmLibraryExtensions_UnBoundOperationsFoundFromIEdmModelFindMethodIsInvalid(operation.Name));
-                }
+            IList<IEdmOperation> operationsFound = operations as IList<IEdmOperation>;

```

Comment:
xactly.

---
In reply to: [493160266](https://github.com/OData/odata.net/pull/1888#discussion_r493160266) [](ancestors = 493160266)

Reply:
as per below, this will usually be a (cached) IList<T>, so should have a significant perf improvement.

---
In reply to: [493150453](https://github.com/OData/odata.net/pull/1888#discussion_r493150453) [](ancestors = 493150453)